pkg/proxyagent/agent/manifests/charts/addon-agent/templates/cluster-service.yaml

```diff
@@ -1,3 +1,4 @@
+{{ if .Values.enableKubeApiProxy }}
```

Comment:
The service is using managed cluster's name as its name, when managed cluster's name starts with a number, it will cause trouble because klusterlet-work-agent can not apply a Service with the name starts with a number. 

```
E1223 09:27:20.700245 1 base_controller.go:270] "ManifestWorkAgent" controller failed to sync "addon-cluster-proxy-deploy-0", err: Service "20pi8725abdjf5lccjlt61dnqidp1bm2" is invalid: metadata.name: Invalid value: "20pi8725abdjf5lccjlt61dnqidp1bm2": a DNS-1035 label must consist of lower case alphanumeric characters or '-', start with an alphabetic character, and end with an alphanumeric character (e.g. 'my-name', or 'abc-123', regex used for validation is '[a-z]([-a-z0-9]*[a-z0-9])?')
```

The newly added flag `enableKubeApiProxy` can remove this service from the Manifestwork when the flag is `false`.